ios/BT/ExposureManager.swift

```diff
@@ -99,35 +90,37 @@ final class ExposureManager: NSObject {
   var localUncompressedURLs = [URL]()
   
   @discardableResult func detectExposures(completionHandler: ((Error?) -> Void)? = nil) -> Progress {
-
-    // Exit if last detection date was < 24 hours ago
-    if !detectionPermitted {
-      completionHandler?(nil)
-    }
     
     let progress = Progress()
-    
+
+    var lastProcessedUrlPath: String = .default
+    var processedFileCount: Int = 0
+
     // Disallow concurrent exposure detection, because if allowed we might try to detect the same diagnosis keys more than once
     guard !detectingExposures else {
       completionHandler?(GenericError.unknown)
       return progress
     }
     
     detectingExposures = true
-    
+
+    // Reset file capacity to 15 if > 24 hours have elapsed since last reset
+    updateRemainingFileCapacity()
+
     func finish(_ result: Result<[Exposure]>) {
       
       cleanup()
-      
+
       if progress.isCancelled {
         detectingExposures = false
         completionHandler?(GenericError.unknown)
       } else {
         switch result {
         case let .success(newExposures):
-          BTSecureStorage.shared.dateLastPerformedExposureDetection = Date()
           BTSecureStorage.shared.exposureDetectionErrorLocalizedDescription = .default
           BTSecureStorage.shared.exposures.append(objectsIn: newExposures)
+          BTSecureStorage.shared.urlOfMostRecentlyDetectedKeyFile = lastProcessedUrlPath
```

Comment:
Looking like we have a of related data being treated separately here? Is there a data clump we want to pull out here such that we save a single object to storage rather than 4 separate entries?

Reply:
The API is a bit misleading, but it's neat - see Persisted.swift - it calls through to setUserState, so per your instinct these are all properties on a single model in the database (UserState)